### DISCUSSION TITLE
Insert key / shortcut stopped working

### DISCUSSION THREAD

**Discussion by SeaofThought** [freeplane/freeplane#493] [Help] [answered]
Hi, I recently upgraded from Freeplane 1.9.10 to 1.10.3, and I couldn't create child nor parent nodes using the Insert shortcut. It still works using Tab.

Is this a bug or a design feature?

**Comment by macmarrum** [comment, marked as answer]
`TAB` is the default in 1.10.3 – please see #443. But you can change it back to `INSERT`. 

To change the hot key, I usually find the item in the menu and Ctrl-click it.

Another option is to first click `Tools->Assign hot key` and then the item, i.e. `Insert->New node->New child node`.

**Reply by Terry046** [reply]
Whoa! Thanks so much for that (i.e., how to change back to INSERT)! I've been using FP for years and never knew I could change the keys, and on my laptop with a 10-key keypad, using INSERT to add a child node was a no-brainer... But after trying the 1.10.3 version, I regretfully reverted to the prior version and was resigned to staying on it (forever?). 

I came to Github to request some/any solution and found it in your post! 

Woohoo and much gratitude!

**Reply by dpolivaev** [reply]
Watch a tutorial
https://youtu.be/WxrAprUAv44

**Reply by echo21** [reply]
Hi @dpolivaev,

I highly support [Loic Joly's request for restoring < ins > as the freeplane default keyboard shortcut for inserting a new child node](https://github.com/freeplane/freeplane/issues/765). I'd like to add another strong argument :

- < ins > key is the de-facto industry standard for inserting new mind map child nodes. Almost all of the commercial and free freeplane alternatives (e.g. MindManagr, MindMeister, XMind, ...) use < ins > instead of < tab > (or at least both of them). To my best knowledge the expensive [MindManager](https://www.mindmanager.com/) by Correl (recently re-branded as [All u do](https://www.alludo.com/)) is the market leader since 1998 (freemind started in 2003 and freeplane forked in 2008). I think that freeplane should not violate de-facto standards.

User who like < tab > more than < ins > can overwrite the default.

**Reply by sifran-github** [reply]
I also support < INS > as default.

Edit: On this issue, I'm just a simple user that likes to use the INS button. I will simply change the keys to INS if TAB is chosen. I just want to note that I don't really know what is the more "correct" way, and I can see the arguments for both.

**Reply by dpolivaev** [reply]
There are some reasons why I changed the key from INS to TAB. I would like you to know them before deciding whether the change should be reverted.

1. On the mac keyboard there is no INS key, and therefore  on macs the key has always been TAB
2. Some people wrote me that their non-Mac notebooks also do not have the INS key
3. TAB key is bigger and easier to find and press than the INS key on the most keyboards
4. Enabling both keys for the same action is possible, but I can not display both keys in the menu so that only one of them would be seen by users. I assumed that seeing which key is responsible must be a no-brainer because all keys are indicated in the menus.

There is a related issue that the TAB key opposite to the INS key does not work during editing, but it is a separate topic; if this is the origin of the problem, it could be solved.

Given this information, could you please write if my arguments could change your opinion and also explain the advantages of the INS key if you still think it is a better choice?

**Reply by sgjohnson1981** [reply]
Standard schmandard. Let Freeplane set the standard!!! Just because something is an industry standard doesn't mean it should be. I changed mine from INS to TAB a long time ago, way before Dimitry changed the default. Not because I was missing the key but because INS isn't within reach of a touch typist like myself, but TAB is. Seems to me the only reason INS was selected as the "industry standard" was the name of the key, i.e. you're "inserting" a sub-node into the tree. I would argue aren't you also "inserting" a node, albeit a sibling, when you hit Enter? Could it be the industry instituted a standard without considering the reasons outlined by Dimitry? Some keyboards don't even have the key.

**Reply by echo21** [reply]
Hi Dimitry ( @dpolivaev ),

thanks for getting back! 

- @1: to my honest opinion the de-facto standard keyboard shortcut for inserting a new child node on Macs seems to be < crtl >< return > but not < tab > . See https://help.mindjet.com/mac/11/MMMac11_Help/EN/pgs2/kbshortcuts.html
`Insert subtopic: (⌘)-Return, Control-Return`

- @1: < tab > key instead has a different purpose on Macs (at least in MindManager for Mac):
`Select next topic / previous topic: Tab / Shift-Tab`

- @2: confirmed

- @3: confirmed

- @4: good to know! So, what about simply adding < ins > key as a second, hidden and un-documented key for inserting a new child node? And it's also perfectly fine that < ins > won't work for this anymore if the user assigns < ins > key to a different command ( < ins > will get unhidden then).

**Comment by greg4b** [comment]
My Windows Notebook HP 830G6 also does not have an INS-key...

Am Mo., 2. Jan. 2023 um 20:10 Uhr schrieb Dimitry Polivaev <
***@***.***>:
